File: bot-api/dotnet/test/src/TeamMessageLimitsTest.cs
using System;
using NUnit.Framework;
using Robocode.TankRoyale.BotApi.Internal;

namespace Robocode.TankRoyale.BotApi.Tests;

[TestFixture]
[Category("Unit")]
public class TeamMessageLimitsTest
{
    [Test]
    public void CountBoundary()
    {
        Assert.DoesNotThrow(() => IntentValidator.ValidateTeamMessage("hello", Constants.MaxNumberOfTeamMessagesPerTurn - 1));
        Assert.Throws<InvalidOperationException>(() => IntentValidator.ValidateTeamMessage("hello", Constants.MaxNumberOfTeamMessagesPerTurn));
    }

    [Test]
    public void UnicodePayloadBoundary()
    {
        Assert.DoesNotThrow(() => IntentValidator.ValidateTeamMessageSize(new string('é', Constants.TeamMessageMaxSize / 2)));
        Assert.Throws<ArgumentException>(() => IntentValidator.ValidateTeamMessageSize(new string('é', Constants.TeamMessageMaxSize / 2 + 1)));
    }

    [Test]
    public void AggregateBoundary()
    {
        Assert.DoesNotThrow(() => IntentValidator.ValidateTeamMessagesSize(new string('x', Constants.TeamMessagesMaxBytesPerTurn)));
        Assert.Throws<ArgumentException>(() => IntentValidator.ValidateTeamMessagesSize(new string('x', Constants.TeamMessagesMaxBytesPerTurn + 1)));
    }
}

File: bot-api/typescript/test/TeamMessageLimits.test.ts
import { describe, expect, test } from "vitest";
import { Constants } from "../src/Constants.js";
import { IntentValidator } from "../src/internal/intentValidator.js";
import { BotException } from "../src/BotException.js";

describe("Unit: team message limits", () => {
  test("accepts 64 messages and rejects the 65th", () => {
    expect(() => IntentValidator.validateTeamMessage("hello", 10)).not.toThrow();
    expect(() => IntentValidator.validateTeamMessage("hello", Constants.MAX_NUMBER_OF_TEAM_MESSAGES_PER_TURN - 1)).not.toThrow();
    expect(() => IntentValidator.validateTeamMessage("hello", Constants.MAX_NUMBER_OF_TEAM_MESSAGES_PER_TURN)).toThrow(BotException);
  });

  test("counts Unicode payload bytes", () => {
    expect(() => IntentValidator.validateTeamMessageSize("é".repeat(Constants.TEAM_MESSAGE_MAX_SIZE / 2))).not.toThrow();
    expect(() => IntentValidator.validateTeamMessageSize("é".repeat(Constants.TEAM_MESSAGE_MAX_SIZE / 2 + 1))).toThrow();
  });

  test("includes the aggregate boundary", () => {
    expect(() => IntentValidator.validateTeamMessagesSize("x".repeat(Constants.TEAM_MESSAGES_MAX_BYTES_PER_TURN))).not.toThrow();
    expect(() => IntentValidator.validateTeamMessagesSize("x".repeat(Constants.TEAM_MESSAGES_MAX_BYTES_PER_TURN + 1))).toThrow();
  });
});
